sklearn/utils/extmath.py

```diff
@@ -75,14 +75,16 @@ def row_norms(X, squared=False):
     array-like
         The row-wise (squared) Euclidean norm of X.
     """
+    xp, _ = get_namespace(X)
+
     if sparse.issparse(X):
         X = X.tocsr()
         norms = csr_row_norms(X)
     else:
-        norms = np.einsum("ij,ij->i", X, X)
+        norms = xp.sum(xp.multiply(X, X), axis=1)
```

Comment:
In other functions we made a `if` statement to keep the old path/numpy code and use the "array api way" for everything else. Not sure but I assume `np.einsum("ij,ij->i", X, X)` is better (less memory?) than using `np.sum(np.multiply(X, X), axis=1)`. So I propose we make a `if` based on the second return value of `get_namespace()`